Automattic/sensei
# Add/import lessons task

### Overview
This PR implements the importing for lessons and quizzes. It also extracts the schema from the model classes to be more easily shared between import and (not yet implemented export) models.

### Changes proposed in this Pull Request
* Introduced `Sensei_Data_Port_Schema` class which defines the schema of every ported entity.
* Moved the existing schema out of the model classes and into seperate subclasses of this new class.
* Introduced `Sensei_Import_Model` class which is a model with import specific functionality. The common functionality of import and export models exist in `Sensei_Data_Port_Model`. These two classes might not be finalized yet but I thought that it is better to do it once we start implementing the export model class.
* Introduced `Sensei_Import_Lesson_Model` which handles lesson and quiz import. This class is a bit bigger than usual because it handles porting of two post types but other than that it is pretty simple.


### Testing instructions

* With this PR all levels can be imported in one go. You can use [this spreadsheet](https://docs.google.com/spreadsheets/d/11Jev2S1VtMdmb7_O9NRXrO56XwRwCtc-g-WVHTPr-1I/edit#gid=0) to extract CSV files for testing. For this PR you should focus on lessons and their quizzes.
* After uploading the files and clicking continue, make sure that background jobs are run.
* Alternatively you can use the CLI tool (i.e. `wp sensei-import --user=wordpress --questions=questions.csv --courses=courses.csv --lessons=lessons.csv`)
* Once the import is completed, check the entities that have been created.
* The state of the jobs can be seen with the query: `SELECT * FROM wp_options WHERE option_name LIKE 'sensei-data-port-job%';`. The state is a JSON object and in it's 'l' array you can find the errors of the job.
